client/pkg/eventlog/log_segment.go

```diff
@@ -171,6 +171,21 @@ func (s *segment) Append(ctx context.Context, event *ce.Event) (int64, error) {
 	return off + s.startOffset, nil
 }
 
+func (s *segment) SyncAppendStream(ctx context.Context, event *ce.Event) (int64, error) {
+	_ctx, span := s.tracer.Start(ctx, "AppendStream")
+	defer span.End()
+
+	b := s.preferSegmentBlock()
+	if b == nil {
+		return -1, errors.ErrNotLeader
+	}
+	off, err := b.SyncAppendStream(_ctx, event)
```

Comment:
will each block create a stream?

Reply:
The current client caches a writable segment and all readable segments. Each segment caches a block and also a stream